src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/sentryApplicationDetails.jsx

```diff
@@ -97,6 +110,70 @@ export default class SentryApplicationDetails extends AsyncView {
     }
   };
 
+  onAddToken = async evt => {
+    evt.preventDefault();
+    const {app, tokens} = this.state;
+    const api = this.api;
+
+    try {
+      const token = await addSentryAppToken(api, app);
+      const newTokens = tokens.concat(token);
+      this.setState({tokens: newTokens});
+    } catch (err) {
+      this.setState({tokens});
+    }
+  };
+
+  onRemoveToken = async (token, evt) => {
+    evt.preventDefault();
+    const {app, tokens} = this.state;
+    const api = this.api;
+    const newTokens = tokens.filter(tok => tok.token !== token.token);
+
+    try {
+      await removeSentryAppToken(api, app, token.token);
+      this.setState({tokens: newTokens});
+    } catch (err) {
+      this.setState({tokens});
```

Comment:
@MeredithAnya I don't think you need the `setState` on error here since the token list is unchanged.